Cython/Compiler/Optimize.py

```diff
@@ -230,6 +337,8 @@ def _optimise_for_loop(self, node, iterable, reversed=False):
             return self._transform_unicode_iteration(node, iterable, reversed=reversed)
 
         # the rest is based on function calls
+        if isinstance(iterable, UtilNodes.ResultRefNode):
+            iterable = iterable.expression # pop the call out of the expression
```

Comment:
OK - I've caught this - 2 spaces before the comment. I'll fix this before merging it but it doesn't seem worth rerunning all the tests right now for the sake of this.